### ISSUE TITLE
Bug: crash when disconnecting monitor

### ISSUE THREAD

**Issue by bernikr** [hass-agent/HASS.Agent#83]
**Describe the bug**
I just switched from the old hass.agent to this fork and encountered the following problem: I'm using my laptop connected to 2 external monitors via an USB-C/Displayport dock. When I disconnect my laptop from the dock hass.agent just crashes and has to be manually restarted. The satellite service however stays running. I was able to reproduce the bug also by just turning one of my monitors off.

From what I can see in the logs this could be related to #33 

**To Reproduce**
1. have hass.agent running on a laptop connected to an external monitor
2. unplug the monitor
3. hass.agent crashes

**Expected behavior**
hass.agent should not crash when unplugging my laptop from its dock

**Misc info (please complete the following information):**
 - Windows build: Windows 11 Version 23H2 (OS Build 22631.3527)
 - Windows' UI language: English US (but German on Windows install)
 - HASS.Agent version: 2.0.1

Please check what's applicable (multiple answers possible):

 - [x] Installed via installer
 - [ ] Installed manually
 - [x] Problem occurs in HASS.Agent
 - [ ] Problem occurs in Satellite Service

**Additional context**
I upgraded from the old hass.agent in the following order:
1. install new hass.agent
2. use the setting migration feature in the setup
3. uninstall the old hass.agent

**Logs**
```
2024-05-03 20:56:54.165 +02:00 [INF] [MAIN] HASS.Agent version: 2.0.1
2024-05-03 20:56:54.172 +02:00 [INF] [MAIN] Extended logging enabled
2024-05-03 20:56:54.177 +02:00 [INF] [SETTINGS] Config storage path: C:\Users\berni\AppData\Local\HASS.Agent\Client\config
2024-05-03 20:56:54.342 +02:00 [INF] [SETTINGS] Configuration loaded
2024-05-03 20:56:54.347 +02:00 [INF] [LOCALIZATION] Selected UI culture: [en] English
2024-05-03 20:56:54.756 +02:00 [INF] [RADIOMGR] Ready
2024-05-03 20:56:56.989 +02:00 [INF] [INTERNALSENSORS] Ready
2024-05-03 20:56:59.997 +02:00 [FTL] [PROGRAM] FirstChanceException: Specified cast is not valid.
System.InvalidCastException: Specified cast is not valid.
   at WindowsDesktop.Interop.IServiceProvider.QueryService(Guid& guidService, Guid& riid)
   at WindowsDesktop.Interop.ComInterfaceAssembly.CreateInstance(Type type, Nullable`1 guidService)
   at WindowsDesktop.Interop.ComInterfaceAssembly.CreateInstance(String comInterfaceName, Guid clsid)
   at WindowsDesktop.Interop.ComWrapperBase`1..ctor(ComInterfaceAssembly assembly, Guid clsid)
   at WindowsDesktop.Interop.Build22621.VirtualDesktopManagerInternal..ctor(ComInterfaceAssembly assembly, ComWrapperFactory factory)
   at WindowsDesktop.Interop.Build22621.VirtualDesktopProvider22621.InitializeCore(ComInterfaceAssembly assembly)
   at WindowsDesktop.Interop.VirtualDesktopProvider.Initialize(ComInterfaceAssembly assembly)
   at WindowsDesktop.VirtualDesktop.InitializeCore()
   at WindowsDesktop.VirtualDesktop.InitializeIfNeeded()
   at WindowsDesktop.VirtualDesktop.Configure()
   at HASS.Agent.Managers.VirtualDesktopManager.Initialize() in D:\a\HASS.Agent\HASS.Agent\src\HASS.Agent\HASS.Agent\Managers\VirtualDesktopManager.cs:line 24
2024-05-03 20:56:59.997 +02:00 [ERR] [VIRTDESKT] Error initializing Virtual Desktop Manager, your Windows version may be unsupported
2024-05-03 20:57:00.065 +02:00 [INF] [SETTINGS_QUICKACTIONS] Loaded 4 entities
2024-05-03 20:57:00.067 +02:00 [INF] [SETTINGS_COMMANDS] Config not found, no entities loaded
2024-05-03 20:57:00.097 +02:00 [INF] [SETTINGS_SENSORS] Loaded 1 entities
2024-05-03 20:57:00.113 +02:00 [INF] [LOCALAPI] Disabled
2024-05-03 20:57:00.114 +02:00 [INF] [MQTT] Identifying as device: NEWYOGA
2024-05-03 20:57:00.121 +02:00 [INF] [HOTKEY] Completed bind for 4 individual quickaction hotkeys
2024-05-03 20:57:00.138 +02:00 [INF] [MQTT] Connecting ..
2024-05-03 20:57:00.158 +02:00 [INF] [SERVICE] Local install path: C:\Program Files\HASS.Agent\Service\
2024-05-03 20:57:00.165 +02:00 [INF] [MEDIA] Disabled
2024-05-03 20:57:00.354 +02:00 [INF] [NOTIFIER] Ready
2024-05-03 20:57:00.411 +02:00 [INF] [MQTT] Connected
2024-05-03 20:57:00.456 +02:00 [INF] [HASS_API] Home Assistant version: 2024.5.1
2024-05-03 20:57:01.160 +02:00 [INF] [HASS_API] System connected with https://smarthome.kralofsky.com
2024-05-03 20:57:02.168 +02:00 [INF] [MQTT] Initial registration completed
2024-05-03 20:57:13.144 +02:00 [FTL] [PROGRAM] FirstChanceException: Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Windows.Forms.ListView.InsertItems(Int32 displayIndex, ListViewItem[] items, Boolean checkHosting)
   at System.Windows.Forms.ListView.ApplyUpdateCachedItems()
   at System.Windows.Forms.ListView.EndUpdate()
   at HASS.Agent.Controls.Service.ServiceSensors.UpdateSensorsList() in D:\a\HASS.Agent\HASS.Agent\src\HASS.Agent\HASS.Agent\Controls\Service\ServiceSensors.cs:line 78
2024-05-03 20:57:44.334 +02:00 [FTL] [PROGRAM] FirstChanceException: Unable to read data from the transport connection: An established connection was aborted by the software in your host machine..
System.IO.IOException: Unable to read data from the transport connection: An established connection was aborted by the software in your host machine..
 ---> System.Net.Sockets.SocketException (10053): An established connection was aborted by the software in your host machine.
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](TIOAdapter adapter)
2024-05-03 20:57:44.341 +02:00 [FTL] [NET] [ConnectionAborted] An established connection was aborted by the software in your host machine.
System.Net.Sockets.SocketException (10053): An established connection was aborted by the software in your host machine.
   at MQTTnet.Implementations.MqttTcpChannel.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at MQTTnet.Adapter.MqttChannelAdapter.ReadFixedHeaderAsync(CancellationToken cancellationToken)
2024-05-03 20:57:44.342 +02:00 [FTL] [NET] [ConnectionAborted] An established connection was aborted by the software in your host machine.
System.Net.Sockets.SocketException (10053): An established connection was aborted by the software in your host machine.
   at MQTTnet.Implementations.MqttTcpChannel.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at MQTTnet.Adapter.MqttChannelAdapter.ReadFixedHeaderAsync(CancellationToken cancellationToken)
2024-05-03 20:57:44.346 +02:00 [FTL] [PROGRAM] FirstChanceException: A task was canceled.
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at MQTTnet.Client.MqttClient.TrySendKeepAliveMessages(CancellationToken cancellationToken)
2024-05-03 20:57:44.347 +02:00 [ERR] [NET] [FirstChanceHttpRequestException.SocketException] An error occurred while sending the request.
2024-05-03 20:57:44.349 +02:00 [ERR] [NET] [FirstChanceHttpRequestException.SocketException] An error occurred while sending the request.
2024-05-03 20:57:44.379 +02:00 [FTL] [PROGRAM] FirstChanceException: A task was canceled.
System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at MQTTnet.Internal.AsyncQueue`1.TryDequeueAsync(CancellationToken cancellationToken)
2024-05-03 20:57:44.552 +02:00 [ERR] [NET] [FirstChanceHttpRequestException.SocketException] An error occurred while sending the request.
2024-05-03 20:57:44.559 +02:00 [ERR] [NET] [FirstChanceHttpRequestException.SocketException] An error occurred while sending the request.
2024-05-03 20:57:44.571 +02:00 [ERR] [NET] [FirstChanceHttpRequestException.SocketException] An error occurred while sending the request.
2024-05-03 20:57:44.788 +02:00 [FTL] [PROGRAM] FirstChanceException: Specified cast is not valid.
System.InvalidCastException: Specified cast is not valid.
   at WindowsDesktop.Interop.IServiceProvider.QueryService(Guid& guidService, Guid& riid)
   at WindowsDesktop.Interop.ComInterfaceAssembly.CreateInstance(Type type, Nullable`1 guidService)
   at WindowsDesktop.Interop.ComInterfaceAssembly.CreateInstance(String comInterfaceName, Guid clsid)
   at WindowsDesktop.Interop.ComWrapperBase`1..ctor(ComInterfaceAssembly assembly, Guid clsid)
   at WindowsDesktop.Interop.Build22621.VirtualDesktopManagerInternal..ctor(ComInterfaceAssembly assembly, ComWrapperFactory factory)
   at WindowsDesktop.Interop.Build22621.VirtualDesktopProvider22621.InitializeCore(ComInterfaceAssembly assembly)
   at WindowsDesktop.Interop.VirtualDesktopProvider.Initialize(ComInterfaceAssembly assembly)
   at WindowsDesktop.VirtualDesktop.InitializeCore()
   at WindowsDesktop.VirtualDesktop.HandleExplorerRestarted()
   at WindowsDesktop.VirtualDesktop.<>c.<.cctor>b__70_0()
   at WindowsDesktop.Utils.ExplorerRestartListenerWindow.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
2024-05-03 20:57:44.789 +02:00 [FTL] [MAIN] ThreadException: Specified cast is not valid.
System.InvalidCastException: Specified cast is not valid.
   at WindowsDesktop.Interop.IServiceProvider.QueryService(Guid& guidService, Guid& riid)
   at WindowsDesktop.Interop.ComInterfaceAssembly.CreateInstance(Type type, Nullable`1 guidService)
   at WindowsDesktop.Interop.ComInterfaceAssembly.CreateInstance(String comInterfaceName, Guid clsid)
   at WindowsDesktop.Interop.ComWrapperBase`1..ctor(ComInterfaceAssembly assembly, Guid clsid)
   at WindowsDesktop.Interop.Build22621.VirtualDesktopManagerInternal..ctor(ComInterfaceAssembly assembly, ComWrapperFactory factory)
   at WindowsDesktop.Interop.Build22621.VirtualDesktopProvider22621.InitializeCore(ComInterfaceAssembly assembly)
   at WindowsDesktop.Interop.VirtualDesktopProvider.Initialize(ComInterfaceAssembly assembly)
   at WindowsDesktop.VirtualDesktop.InitializeCore()
   at WindowsDesktop.VirtualDesktop.HandleExplorerRestarted()
   at WindowsDesktop.VirtualDesktop.<>c.<.cctor>b__70_0()
   at WindowsDesktop.Utils.ExplorerRestartListenerWindow.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
```

**Comment by amadeo-alex** [comment]
Hello,

Could you please try this https://github.com/hass-agent/HASS.Agent/releases/tag/2.1.0-beta1 ?
The one referenced in issue you've linked is this one https://github.com/hass-agent/HASS.Agent/releases/tag/2.0.2-beta2 although I'd recommend going with 2.1.0-beta1 for additional fixes/features.

**Comment by bernikr** [comment]
I've realized in the meantime that the referenced issue was only fixed in the beta and tried out 2.1.0-beta1 and can confirm that it seems to be working now.

Sorry for wasting your time^^

**Comment by amadeo-alex** [comment]
No worries, always could have been something we missed, glad that it's working :D
